cmd/kubectl-gadget/utils/trace.go

```diff
@@ -41,9 +45,39 @@ import (
 
 const (
 	GADGET_OPERATION = "gadget.kinvolk.io/operation"
+	TRACE_ID         = "trace-id"
 	traceTimeout     = 2 * time.Second
 )
 
+// TraceConfig is used to contain information used to create a trace.
+type TraceConfig struct {
+	// GadgetName is gadget name, e.g. socket-collector.
+	GadgetName string
+	// Operation is the trace operation, e.g. start to start the tracing.
+	Operation string
+	// NodeName is the name of the node to trace as given by the '--node' option.
+	// If empty, all nodes within the namespace will be traced.
+	NodeName string
+	// UseAllNamespaces indicates if the tracing will occur on all namespaces or
+	// on current namespace otherwise.
+	UseAllNamespaces bool
+	// PodName is the name of the pod to trace as given by the '-p' option.
+	// If empty, all pods within the namespace will be traced.
+	PodName string
+	// ContainerName is the name of the container to trace as given by the '-c'
+	// option.
+	// If empty, all containers within the namespace will be traced.
+	ContainerName string
+	// Labels contains labels given with the -l option.
+	Labels string
+	// OutputMode is output mode given by the user on CLI.
+	// If this field is empty, we default to "table" print.
+	// Otherwise, the only correct value is "json".
+	OutputMode string
+	// Verbose is used to print more information.
+	Verbose bool
+}
```

Comment:
The problem of setting it from the caller is that every gadget needs to do it while with the previous code, it was computed inside the shared code (`utils/trace.go`). I think Francis was following the same idea of the [utils.CommonFlags](https://github.com/kinvolk/inspektor-gadget/blob/02abdae390abdfb2889e9d52e0b7b2ef76e72c2c/cmd/kubectl-gadget/utils/flags.go#L49-L77) which does not have the namespace field.

BTW, why don't we just add the full `utils.CommonFlags` data structure inside `TraceConfig` instead of adding each field separately? In the end, [the only field](https://github.com/kinvolk/inspektor-gadget/blob/3852ae762c8e12785fd2ab49043ac1f9f96a330f/cmd/kubectl-gadget/seccompadvisor.go#L80-L91) that the gadgets will not use is the CustomColumns but I think it is an acceptable price to pay to keep things simpler, don't you think?

Reply:
> The problem of setting it from the caller is that every gadget needs to do it while with the previous code

The caller is already setting other fields, so one more should not make a big difference. 

> BTW, why don't we just add the full utils.CommonFlags data structure inside TraceConfig instead of adding each field separately?

Yes, that's a very good idea. I think it's already done at the time I'm commenting.